src/vm/diagnosticserver.cpp

```diff
@@ -37,43 +36,51 @@ static DWORD WINAPI DiagnosticsServerThread(LPVOID lpThreadParameter)
         STRESS_LOG2(LF_DIAGNOSTICS_PORT, LL_WARNING, "warning (%d): %s.\n", code, szMessage);
     };
 
-    while (true)
+    EX_TRY
     {
-        // FIXME: Ideally this would be something like a std::shared_ptr
-        IpcStream *pStream = pIpc->Accept(LoggingCallback);
-        if (pStream == nullptr)
-            continue;
-
-        // TODO: Read operation should happen in a loop.
-        uint32_t nNumberOfBytesRead = 0;
-        MessageHeader header;
-        bool fSuccess = pStream->Read(&header, sizeof(header), nNumberOfBytesRead);
-        if (!fSuccess || nNumberOfBytesRead != sizeof(header))
+        while (true)
         {
-            delete pStream;
-            continue;
-        }
+            // FIXME: Ideally this would be something like a std::shared_ptr
+            IpcStream *pStream = pIpc->Accept(LoggingCallback);
+            if (pStream == nullptr)
+                continue;
+
+            // TODO: Read operation should happen in a loop.
+            uint32_t nNumberOfBytesRead = 0;
+            MessageHeader header;
+            bool fSuccess = pStream->Read(&header, sizeof(header), nNumberOfBytesRead);
+            if (!fSuccess || nNumberOfBytesRead != sizeof(header))
+            {
+                delete pStream;
+                continue;
+            }
 
-        // TODO: Dispatch thread worker.
-        switch (header.RequestType)
-        {
-        case DiagnosticMessageType::EnableEventPipe:
-            EventPipeProtocolHelper::EnableFileTracingEventHandler(pStream);
-            break;
+            switch (header.RequestType)
+            {
+            case DiagnosticMessageType::EnableEventPipe:
+                EventPipeProtocolHelper::EnableFileTracingEventHandler(pStream);
+                break;
 
-        case DiagnosticMessageType::DisableEventPipe:
-            EventPipeProtocolHelper::DisableFileTracingEventHandler(pStream);
-            break;
+            case DiagnosticMessageType::DisableEventPipe:
+                EventPipeProtocolHelper::DisableFileTracingEventHandler(pStream);
+                break;
 
-        case DiagnosticMessageType::StreamEventPipe:
-            EventPipeProtocolHelper::AttachTracingEventHandler(pStream);
-            break;
+            case DiagnosticMessageType::StreamEventPipe:
+                EventPipeProtocolHelper::AttachTracingEventHandler(pStream);
+                break;
 
-        default:
-            LOG((LF_DIAGNOSTICS_PORT, LL_WARNING, "Received unknow request type (%d)\n", header.RequestType));
-            break;
+            default:
+                LOG((LF_DIAGNOSTICS_PORT, LL_WARNING, "Received unknow request type (%d)\n", header.RequestType));
+                break;
+            }
         }
     }
+    EX_CATCH
+    {
+        LOG((LF_DIAGNOSTICS_PORT, LL_ERROR, "Exception caught in diagnostic thread. Leaving thread now.\n"));
```

Comment:
I'd recommend STRESS_LOG so that we can diagnose the failure if it occurs in retail builds.